### ISSUE TITLE
Sauce Labs and browser coverage

### ISSUE THREAD

**Issue by matthew-dean** [less/less-meta#21]
Does anyone know anything at all about Sauce Labs? I've been trying to update the tests so that
1. The browsers are updated to latest versions
2. There's more comprehensive coverage of browsers currently in use, 

If we could sufficiently prove that Less.js worked in all common browsers, there's no reason we couldn't say it's production-ready for browser use as well (even though we could still recommend pre-compiling for performance reasons). Compiling-in-browser is much more common than when Less.js started because today's JavaScript engines are so fast (plus regular increases in CPU performance).

I tried to update settings on the 3.x branch, but I wasn't able to get test results to display at https://saucelabs.com/u/less, nor could I figure out how to generate a graphic for test results. Somehow there's a way to have Travis push stuff to Sauce Labs, but after a day of working at it, I couldn't get it to work. I think @lukeapage set it up originally, so Luke, if you have a second to explain this, or if anyone has Sauce Labs experience, that help is currently needed.

**Comment by SomMeri** [comment]
Is this issue still relevant? I click sauce labs links and there seem to be some test results.

**Comment by matthew-dean** [comment]
@SomMeri Somewhat. Another contributor got Sauce Labs working, but I want to track down the remaining errors that cause non-compliance in browsers. As well, I've been trying to reduce the number of html files that have to be run for browser tests, so they can execute faster. I found that while the Less API is technically callable in browser, it's harder to manage than it needs to be, and errors / warnings in browser are not the result of render calls, but the result of magic code that only runs once. In other words, you can't call the API and pass results to the browser in the same way as what happens with `<link />` tags. This is related to: https://github.com/less/less.js/issues/2951

So, first: can we get 100% browser coverage? That's my goal for 3.0.

**Comment by matthew-dean** [comment]
I spent about... I don't know, 80 hours on this, figuring out Sauce and trying to optimize the setup. Unfortunately, the tests are not easy to optimize since less in the browser expects only to be run against a `<link>` tag that's in the DOM from the start. So that was a dead end.

However, I was able to change the test template through a lot of trial and error until it stopped erroring out in IE8 and Chrome on Windows (and iOS Safari). Less works in those environments, it's just that the tests didn't run. So, milestone achieved!
